providers/implementations/keymgmt/ec_kmgmt.c

```diff
@@ -231,10 +232,21 @@ int otherparams_to_params(const EC_KEY *ec, OSSL_PARAM_BLD *tmpl,
                           OSSL_PARAM params[])
 {
     int ecdh_cofactor_mode = 0;
+    int ec_param_encoding = 0;
+    const EC_GROUP *group = EC_KEY_get0_group(ec);
 
     if (ec == NULL)
         return 0;
 
+    if (group != NULL) {
+        ec_param_encoding =
+            (EC_GROUP_get_asn1_flag(group) & OPENSSL_EC_NAMED_CURVE) ? 0 : 1;
+        if (!ossl_param_build_set_int(tmpl, params,
```

Comment:
I don't think this should be an _int_.  A string is used below for the same param.

Reply:
Reworked..